pygments/lexers/mcfunction.py

```diff
@@ -0,0 +1,272 @@
+"""
+    pygments.lexers.mcfunction
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Lexers for MCFunction and related languages.
+
+    :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, default, include, using
+from pygments.token import (Comment, Error, Keyword, Literal, Name, Number,
+                            Operator, Punctuation, String, Text, Token,
+                            Whitespace)
+
+
+__all__ = ['SNBTLexer', 'MCFunctionLexer']
+
+
+class SNBTLexer(RegexLexer):
+    """Lexer for stringified NBT, a data format used in Minecraft
+
+
+    .. versionadded:: ?
+    """
+
+    name = "SNBT"
+    url = ""
+    aliases = ["snbt"]
+    filenames = ["*.snbt"]
+    mimetype = ["text/snbt"]
+
+    tokens = {
+        "root": [
+            # We only look for the open bracket here since square bracket
+            #  is only valid in NBT pathing (which is a mcfunction idea).
+            (r"(?=\{)", Punctuation, "compound"),
+        ],
+
+        "whitespace": [
+            (r"\s+", Whitespace),
+        ],
+
+        "operators": [
+            (r"[,:;]", Punctuation),
+        ],
+
+        "literals": [
+            (r"(true|false)", Keyword.Constant),
+            (r"(-?\d*\.\d+[fFdD]?)", Number.Float),
+            (r"(-?\d+[bBsSlLfFdD]?)", Number.Integer),
+            (r"(-?\d+[eE]-?\d+)", Number.Float),
```

Comment:
Does the language not support `1.5e5`?

Reply:
Language does support this, I just completely forgot! Only supports it in valid SNBT though.

Reply:
That's not a big deal -- usually, Pygments does not attempt to validate the input, just to lex it correctly.